### PR TITLE
mongo scaler support

### PR DESCRIPTION
**Pull request by NUCsimple**
_Provide a description of what has been changed_

### Checklist

- [ ] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Tests have been added
- [ ] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs
- [ ] Changelog has been updated

Fixes #
add mongo support

### CODE REVIEW

File: pkg/scalers/mongo.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+	if val, ok := config.TriggerMetadata["collection"]; ok {
+		meta.collection = val
+	} else {
+		return nil, fmt.Errorf("no collection given")
+	}
+
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no queryFile given")
```
Comment (ringtail) [inline pkg/scalers/mongo.go:97]: change queryFile->query statement

File: pkg/scalers/mongo.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	}
+
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("queryValue parsing error %v", err.Error())
```
Comment (ringtail) [inline pkg/scalers/mongo.go:103]: failed to parse queryValue,because of %v

File: pkg/scalers/mongo.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+	if val, ok := config.TriggerMetadata["dbName"]; ok {
+		meta.dbName = val
+	} else {
+		return nil, fmt.Errorf("no dbName given")
+	}
+
+	// Resolve connectionString
+	if config.AuthParams["connectionString"] != "" {
+		meta.connectionString = config.AuthParams["connectionString"]
+	} else if config.TriggerMetadata["connectionStringFromEnv"] != "" {
```
Comment (ringtail) [inline pkg/scalers/mongo.go:119]: if v,ok := config.TriggerMetadata["connectionStringFromEnv"]; ok {
     meta.connectionString = config.ResolvedEnv[v]
}

File: pkg/scalers/mongo.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+		if val, ok := config.TriggerMetadata["username"]; ok {
+			meta.username = val
+		} else {
+			return nil, fmt.Errorf("no username given")
+		}
+		//get password from env or authParams
+		if config.AuthParams["password"] != "" {
+			meta.password = config.AuthParams["password"]
+		} else if config.TriggerMetadata["passwordFromEnv"] != "" {
```
Comment (ringtail) [inline pkg/scalers/mongo.go:142]: same as prev section

File: pkg/scalers/mongo.go [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+		}
+	}
+
+	return &meta, nil
+}
+
+func (s *mongoScaler) IsActive(ctx context.Context) (bool, error) {
+	result, err := s.getQueryResult()
+	if err != nil {
+		mongoLog.Error(err, fmt.Sprintf("quey Mongo err: %v", err))
```
Comment (ringtail) [inline pkg/scalers/mongo.go:157]: failed to query MongoDB, because of %v

File: pkg/scalers/mongo.go [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+func NewMongoScaler(config *ScalerConfig) (Scaler, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), mongoDefaultTimeOut)
+	defer cancel()
+
+	meta, err := parseMongoMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Mongo metadata: %v", err)
+	}
+
+	connStr := mongoMetadataToConnectionStr(meta)
```
Comment (ringtail) [inline pkg/scalers/mongo.go:61]: no need to convert. you can convert the params in parseMongoMetadata func

File: pkg/scalers/mongo.go [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+	}
+	return connStr
+}
+
+// Close disposes of Mongo connections
+func (s *mongoScaler) Close() error {
+	if s.client != nil {
+		err := s.client.Disconnect(context.TODO())
+		if err != nil {
+			mongoLog.Error(err, fmt.Sprintf("error closing mongo client: %v", err))
```
Comment (ringtail) [inline pkg/scalers/mongo.go:183]: failed to close MongoDB client,because of %v

File: pkg/scalers/mongo.go [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+// getQueryResult query mongoDB by meta.query
+func (s *mongoScaler) getQueryResult() (int, error) {
+	var (
+		results []MongoFields
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), mongoDefaultTimeOut)
+	defer cancel()
+
+	filter, err := kedautil.Json2BsonDoc(s.metadata.query)
```
Comment (ringtail) [inline pkg/scalers/mongo.go:200]: move util from keda util to mongo.go

**Comment by NUCsimple** [comment]
Pls review again, thanks

**Review by ringtail — APPROVED** [review]

**Pull request merged**